Introduce cross-cluster replication API docs (#34726)
elastic/elasticsearch: docs/reference/ccr/apis/follow/get-follow-stats.asciidoc

following tasks associated with each shard for the specified indices.

==== Request

//////////////////////////

[source,js]
--------------------------------------------------
PUT /follower_index/_ccr/follow
{
  "leader_cluster" : "leader_cluster",
  "leader_index" : "leader_index"
}
--------------------------------------------------
// CONSOLE
// TESTSETUP
// TEST[setup:leader_cluster_and_leader_index]

[source,js]
--------------------------------------------------
POST /follower_index/_ccr/pause_follow
--------------------------------------------------
// CONSOLE
// TEARDOWN

//////////////////////////

[source,js]
--------------------------------------------------
GET /_ccr/stats
--------------------------------------------------
// CONSOLE
// TEST

Here, all follower indices will have their follower stats returned.

[source,js]
--------------------------------------------------
GET /<index>/_ccr/stats
--------------------------------------------------
// CONSOLE
// TEST[s/<index>/follower_index/]

Here, `<index>` is a comma-delimited list of index patterns that will be matched
against follower indices to retrieve follower stats for.

> 

Reviewer: ```suggestion

==== Path Parameters

`index` ::
  (string) A comma-delimited list of index patterns.
```

==== Results

This API returns the following information:

`indices`::

>   (array) an array of follower index statistics

Reviewer: ```suggestion
  (array) An array of follower index statistics
```

The `indices` array consists of objects containing two fields:

`indices[].index`::

>   (string) the name of the follower index

Reviewer: ```suggestion
  (string) The name of the follower index
```

`indices[].shards`::

>   (array) an array of shard-level following task statistics

Reviewer: ```suggestion
  (array) An array of shard-level following task statistics
```

The `shards` array consists of objects containing the following fields:

`indices[].shards[].leader_cluster`::

>   (string) the <modules-remote-clusters,remote cluster>> containing the leader

Reviewer: ```suggestion
  (string) The <modules-remote-clusters,remote cluster>> containing the leader
```

  index

`indices[].shards[].leader_index`::

>   (string) the name of the index in the leader cluster being followed

Reviewer: ```suggestion
  (string) The name of the index in the leader cluster being followed
```

`indices[].shards[].follower_index`::

>   (string) the name of the follower index

Reviewer: ```suggestion
  (string) The name of the follower index
```

`indices[].shards[].shard_id`::

>   (integer) the numerical shard ID, with values from 0 to one less than the

Reviewer: ```suggestion
  (integer) The numerical shard ID, with values from 0 to one less than the
```

  number of replicas

`indices[].shards[].leader_global_checkpoint`::

>   (long) the current global checkpoint on the leader known to the follower task

Reviewer: ```suggestion
  (long) The current global checkpoint on the leader known to the follower task
```

`indices[].shards[].leader_max_seq_no`::

>   (long) the current maximum sequence number on the leader known to the follower

Reviewer: ```suggestion
  (long) The current maximum sequence number on the leader known to the follower
```

  task

`indices[].shards[].follower_global_checkpoint`::

>   (long) the current global checkpoint on the follower; the difference between the

Reviewer: ```suggestion
  (long) The current global checkpoint on the follower. The difference between the
```

  `leader_global_checkpoint` and the `follower_global_checkpoint` is an
  indication of how much the follower is lagging the leader

`indices[].shards[].follower_max_seq_no`::

>   (long) the current maximum sequence number on the follower

Reviewer: ```suggestion
  (long) The current maximum sequence number on the follower
```

`indices[].shards[].last_requested_seq_no`::

>   (long) the starting sequence number of the last batch of operations requested

Reviewer: ```suggestion
  (long) The starting sequence number of the last batch of operations requested
```

  from the leader

`indices[].shards[].number_of_concurrent_reads`::

>   (integer) the number of active fetches

Reviewer: ```suggestion
  (integer) The number of active fetches
```

`indices[].shards[].number_of_concurrent_writes`::

>   (integer) the number of active writes

Reviewer: ```suggestion
  (integer) The number of active writes
```

`indices[].shards[].number_of_queued_writes`::

>   (integer) the number of write operations queued on the follower

Reviewer: ```suggestion
  (integer) The number of write operations queued on the follower
```

`indices[].shards[].mapping_version`::

>   (long) the mapping version on the follower

Reviewer: ```suggestion
  (long) The mapping version on the follower
```

`indices[].shards[].total_fetch_time_millis`::

>   (long) the total time fetches were outstanding, measured from the time a fetch

Reviewer: ```suggestion
  (long) The total time fetches were outstanding, measured from the time a fetch
```

  was sent to the leader to the time a reply was returned to the follower

`indices[].shards[].total_fetch_leader_time_millis`::

>   (long) the total time fetches spent executing on the remote cluster

Reviewer: ```suggestion
  (long) The total time fetches spent executing on the remote cluster
```

`indices[].shards[].number_of_successful_fetches`::

>   (long) the number of successful fetches

Reviewer: ```suggestion
  (long) The number of successful fetches
```

`indices[].shards[].number_of_failed_fetches`::

>   (long) the number of failed fetches

Reviewer: ```suggestion
  (long) The number of failed fetches
```

`indices[].shards[].operations_received`::

>   (long) the total number of operations received from the leader

Reviewer: ```suggestion
  (long) The total number of operations received from the leader
```

`indices[].shards[].total_transferred_bytes`::

>   (long) the total of transferred bytes received from the leader (note this is

Reviewer: ```suggestion
  (long) The total of transferred bytes received from the leader. Note this is
```

>   only an estimate and does not account for compression if enabled)

Reviewer: ```suggestion
  only an estimate and does not account for compression if enabled
```

`indices[].shards[].total_index_time_millis`::

>   (long) the total time spent indexing on the follower

Reviewer: ```suggestion
  (long) The total time spent indexing on the follower
```

`indices[].shards[].number_of_successful_bulk_operations`::

>   (long) the number of bulk operations executed on the follower

Reviewer: ```suggestion
  (long) The number of bulk operations executed on the follower
```

`indices[].shards[].number_of_failed_bulk_operations`::

>   (long) the number of failed operations executed on the follower

Reviewer: ```suggestion
  (long) The number of failed operations executed on the follower
```

`indices[].shards[].number_of_operations_indexed`::

>   (long) the number of operations indexed on the follower

Reviewer: ```suggestion
  (long) The number of operations indexed on the follower
```

`indices[].shards[].fetch_exceptions`::

>   (array) an array of objects representing failed fetches

Reviewer: ```suggestion
  (array) An array of objects representing failed fetches
```

The `fetch_exceptions` array consists of objects containing the following
fields:

`indices[].shards[].fetch_exceptions[].from_seq_no`::

>   (long) the starting sequence number of the batch requested from the leader

Reviewer: ```suggestion
  (long) The starting sequence number of the batch requested from the leader
```

`indices[].shards[].fetch_exceptions[].retries`::

>   (integer) the number of times the batch has been retried

Reviewer: ```suggestion
  (integer) The number of times the batch has been retried
```

`indices[].shards[].fetch_exceptions[].exception`::

>   (object) represents the exception that caused the fetch to fail

Reviewer: ```suggestion
  (object) Represents the exception that caused the fetch to fail
```

Continuing with the fields from `shards`:

`indices[].shards[].time_since_last_fetch_millis`::

>   (long) the number of milliseconds since a fetch request was sent to the

Reviewer: ```suggestion
  (long) The number of milliseconds since a fetch request was sent to the
```

>   leader; note that when the follower is caught up to the leader, this number

Reviewer: ```suggestion
  leader. Note that when the follower is caught up to the leader, this number
```

>   will increase up to the configured `poll_timeout` at which point another fetch

Reviewer: ```suggestion
  increases up to the configured `poll_timeout` at which point another fetch
```

>   request will be sent to the leader

Reviewer: ```suggestion
  request is sent to the leader
```

`indices[].fatal_exception`::

>   (object) an object representing a fatal exception that cancelled the following

Reviewer: ```suggestion
  (object) An object representing a fatal exception that cancelled the following
```

>   task; in this situation, the following task must be resumed manually with the

Reviewer: ```suggestion
  task. In this situation, the following task must be resumed manually with the
```

  <<ccr-post-resume-follow,resume follower API>>

==== Example

> 

Reviewer: ```suggestion

This example retrieves follower stats: 
```
